src/scribe_data/checkquery.py

```diff
@@ -0,0 +1,442 @@
+
+    return False
+
+
+def all_queries() -> list[QueryFile]:
+    """All the SPARQL queries in, and below, 'Scribe-Data/'.
+
+    Returns:
+        list[QueryFile]: the SPARQL query files.
+    """
+    parts = Path(__file__).resolve().parts
+    prj_root_idx = parts.index(PROJECT_ROOT)
+    prj_root = str(Path(*parts[: prj_root_idx + 1]))
+
+    queries: list[QueryFile] = []
+
+    for root, _, fnames in os.walk(prj_root):
+        for fname in fnames:
+            fpath = Path(root, fname)
+            if fpath.suffix == ".sparql":
+                queries.append(QueryFile(fpath))
+
+    return queries
+
+
+def changed_queries() -> Optional[list[QueryFile]]:
+    """Find all the SPARQL queries that have changed.
+
+    Includes new queries.
+
+    Returns:
+        Optional[list[QueryFile]]: list of changed/new SPARQL queries or None if error.
+    """
+
+    result = subprocess.run(
+        (
+            "git",
+            "status",
+            "--short",
+        ),
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        universal_newlines=True,
+        check=False,
+    )
+
+    if result.returncode != EXIT_SUCCESS:
+        print(f"ERROR: {result.stderr}", file=sys.stderr)
+        return None
+
+    # example 'result.stdout'
+    #
+    # M a/b/c/changed.sparql
+    # ?? ../new.sparql
+
+    changed_files = [
+        Path(norm_line.split(maxsplit=1)[1]).resolve()
+        for line in result.stdout.split("\n")
+        if (norm_line := line.strip())
+    ]
+
+    return [QueryFile(fpath) for fpath in changed_files if fpath.suffix == ".sparql"]
+
+
+def sparql_context(url: str) -> SPARQL.SPARQLWrapper:
+    """Configure a SPARQL context.
+
+    A context allows the execution of SPARQL queries.
+
+    Args:
+        url (str): a valid URL of a SPARQL endpoint.
+
+    Returns:
+        SPARQLWrapper: the context.
+    """
+    context = SPARQL.SPARQLWrapper(url)
+    context.setReturnFormat(SPARQL.JSON)
+    context.setMethod(SPARQL.POST)
+
+    return context
+
+
+def execute(
+    query: QueryFile, limit: int, context: SPARQL.SPARQLWrapper, tries: int = 3
+) -> dict:
+    """Execute a SPARQL query in a given context.
+
+    Args:
+        query (QueryFile): the SPARQL query to run.
+        limit (int): the maximum number of results a query should return.
+        context (SPARQLWrapper): the SPARQL context.
+        tries (int): the maximum number of times the query should be executed
+                    after failure.
+
+    Returns:
+        dict: results of the query.
+    """
+
+    def delay_in_seconds() -> int:
+        """How long to wait, in seconds, between executing repeat queries."""
+        return int(math.ceil(10.0 / math.sqrt(tries)))
+
+    if tries <= 0:
+        raise QueryExecutionException("Failed too many times.", query)
+
+    try:
+        context.setQuery(query.load(limit))
+        result = context.query().convert()
+        return result if result else []
```

Comment:
On else branch, should return empty dictionary `{}`

Reply:
I guess for a lof of the sparqlwrapper methods we have camelCase, so maybe making this change makes sense? I'm fine either way though, and thank you for doing the research into this!